test/integration/admin/api/settings_controller_test.rb

```diff
@@ -61,6 +61,14 @@ def setup
 
     put admin_api_settings_path(format: :json), params: { access_token: token, enforce_sso: true }
 
+    assert_response :unprocessable_entity
+    error = JSON.parse(response.body)['errors']['enforce_sso']
+    assert_equal ["Password-based authentication could not be disabled. No published authentication providers."], error
```

Comment:
maybe also check validation fails for existing auth provider but not published

Reply:
I will add this in the validator's test. Here I don't think we need to test all the different conditions, we just need to check the response in case of validation error.

I even thought here just stubbing the validator's response, but I thought that the behavior is less clear this way.

Reply:
Addressed here: https://github.com/3scale/porta/pull/3822/commits/109cb2bae70843b56afefb55c82309ed2b5d18ba#diff-87ba48b34ff4f5582358bc12da21bf6cbdd99d832eb0378180b85a334e22578dR16-R21